lumicks/pylake/force_calibration/detail/salty_water.py

```diff
@@ -0,0 +1,222 @@
+        )
+
+    if not np.all(pressure <= 35):
+        raise ValueError(f"{which} function is only valid for p <= 35 MPa, you provided {pressure}")
+
+    if not np.all(molality <= 6.0):
+        raise ValueError(
+            f"{which} function is only valid for m <= 6.0 mol/kg NaCl, you provided {molality}"
+        )
+
+
+def _density_of_salt_solution(temperature, molality, pressure):
+    """Determine the density of water with NaCl.
+
+    This model is based on [1]_.
+
+    Parameters
+    ----------
+    temperature : array_like
+        Temperature in C
+    molality : float
+        Molality NaCl [mol/kg]
+    pressure : float, optional
+        Pressure (default: 0.101325) [MPa].
+
+    Raises
+    ------
+    ValueError
+        When the provided values are outside the valid range of this model. The valid ranges are:
+        Temperature (20°C <= T < 150°C), pressure <= 35 MPa and molality <= 6 mol/kg.
+
+    References
+    ----------
+    .. [1] Kestin, J., Khalifa, H. E., & Correia, R. J. (1981). Tables of the dynamic and
+       kinematic viscosity of aqueous NaCl solutions in the temperature range 20–150 C and
+       the pressure range 0.1–35 MPa. Journal of physical and chemical reference data, 10(1),
+       71-88.
+    """
+    import scipy.constants
+
+    _check_salt_model_validity("Density", temperature, pressure, molality)
+
+    temperature_k = scipy.constants.convert_temperature(temperature, "C", "K")
+    mass_nacl = molality * 58.4428 / 1000  # kg NaCl
+    mass_fraction = mass_nacl / (1 + mass_nacl)  # mass solute / mass total
+
+    powers_ab = np.arange(-2, 3, 1)
+    powers_other = np.arange(0, 3, 1)
+
+    # All polynomials only depend on temperature
+    def poly(powers, coefficients):
+        return _poly(temperature_k, powers, coefficients)
+
+    # See page 73 of [1].
+    a_coeff = [1.006741e2, -1.127522, 5.916365e-3, -1.035794e-5, 9.270048e-9]
+    b_coeff = [1.042948, -1.1933677e-2, 5.307535e-5, -1.0688768e-7, 8.492739e-11]
+
+    c_coeff = [1.23268e-9, -6.861928e-12, 0]
+    d_coeff = [-2.5166e-3, 1.11766e-5, -1.70552e-8]
+    e_coeff = [2.84851e-3, -1.54305e-5, 2.23982e-8]
+    f_coeff = [-1.5106e-5, 8.4605e-8, -1.2715e-10]
+    g_coeff = [2.7676e-5, -1.5694e-7, 2.3102e-10]
+    h_coeff = [6.4633e-8, -4.1671e-10, 6.8599e-13]
+
+    terms = [
+        poly(powers_ab, a_coeff),
+        -poly(powers_ab, b_coeff) * pressure,
+        -poly(powers_other, c_coeff) * pressure**2,
+        mass_fraction * poly(powers_other, d_coeff),
+        (mass_fraction**2) * poly(powers_other, e_coeff),
+        -mass_fraction * poly(powers_other, f_coeff) * pressure,
+        -(mass_fraction**2) * poly(powers_other, g_coeff) * pressure,
+        -0.5 * poly(powers_other, h_coeff) * pressure**2,
+    ]
+
+    return 1.0 / np.sum(terms, axis=0)
+
+
+def water_molarity_to_molality(molarity, temperature, pressure, mol_weight=58.4428):
```

Comment:
A couple points here, all kind of related
- I find this name confusing because you're not calculating the concentration of water. I would leave it generic, jus `molarity_to_molality`. 
- Because it's generic I think it might also be a good idea to not give a default value to `mol_weight` but rather pass in the MW of NaCl in the public `density_of_water` function (because that one is specifically for correcting for NaCl concentration)
- If you don't mind, I would change `mol_weight` to `molecular_weight` or at least `molec_weight`. This might be personal, but the chemist in me sees "mol" and immediately equates this to "moles". Since it's only used once, I think it's ok to have the longer name

Reply:
- I find this name confusing because you're not calculating the concentration of water. I would leave it generic, jus molarity_to_molality.
Good point. I'll change it.

- Because it's generic I think it might also be a good idea to not give a default value to `mol_weight` but rather pass in the MW of NaCl in the public `density_of_water` function (because that one is specifically for correcting for NaCl concentration)
I've adjusted it. But technically they're not that generic. Since they internally depend on `_density_of_salt_solution`, they are specific to this solution. That said, changing it will make it easier in the future to possibly pass in a custom density function.

- If you don't mind, I would change mol_weight to molecular_weight or at least molec_weight. This might be personal, but the chemist in me sees "mol" and immediately equates this to "moles". Since it's only used once, I think it's ok to have the longer name
Yeah, makes sense. I'll update it.